Bottom: de6a6b78e284bea8263f34d5d05369232c7c1166
Top:    2b4fc3ae796cdcb70d47a44d21b04958b48ab618
Author: Peter Clifton <peter@clifton-electronics.co.uk>
Date:   2015-01-09 23:54:41 +0000

Fix to ParseGroupString()


---

diff --git a/src/misc.c b/src/misc.c
index 31021c9..b01a5ae 100644
--- a/src/misc.c
+++ b/src/misc.c
@@ -1101,7 +1101,14 @@ ParseGroupString (char *group_string, LayerGroupType *LayerGroup, int *LayerN)
           /* ignore white spaces and check for separator */
           while (*s && isspace ((int) *s))
             s++;
-          if (!*s || *s == ':')
+
+          /* Backup so the outer loop terminates on the NULL termination */
+          if (*s == '\0') {
+            s--;
+            break;
+          }
+
+          if (*s == ':')
             break;
         }
       LayerGroup->Number[group] = member;
